gson/src/main/java/com/google/gson/GsonBuilder.java

```diff
@@ -325,6 +327,32 @@ public GsonBuilder setFieldNamingStrategy(FieldNamingStrategy fieldNamingStrateg
     return this;
   }
 
+  /**
+   * Configures Gson to apply a specific number strategy during deserialization for {@link Object}.
```

Comment:
Maybe "deserialization of" / "deserialization as" instead of "for" (same for `setNumberToNumberStrategy`)?